### PR TITLE
[DDW-933] Implement Applitools visual regression testing

### PR DESCRIPTION
**Pull request by tarynbillie**
**This PR integrates visual regression testing with a framework called Applitools that utilizes Storybook.**

Visual testing is a form of regression test that ensures that screens that were correct have not changed unexpectedly. This visual testing is done by saving screen snapshots at key points called _checkpoints_. These screenshots are then compared to previously stored baseline images, and any significant differences are reported, ignoring differences that a human couldn't perceive.

To familiarize yourself more in-depth with this automated visual regression testing I recommend checking out the [Storybook docs](https://github.com/storybookjs/storybook/blob/master/docs/src/pages/testing/automated-visual-testing/index.md). I also found the [Applitools overview docs](https://applitools.com/docs/topics/overview/overview-visual-testing.html) to be super helpful in understanding the core process. Applitools also provides an [SDK](https://applitools.com/tutorials/storybook-react.html#how-it-works) which will be similar to the checklist I've created below but in greater detail. I recommend that you set up your [Github CI integration](https://applitools.com/docs/topics/integrations/github-integration.html) if you want the test to run automatically on Pull Requests.

---
### Getting started with Applitools

**Your Applitools account is free to use for open-source software!**

- [ ] Create an Applitools account
[https://applitools.com/users/register](url)
- [ ] Click the profile icon on the top right corner menu
Screen Shot 2019-09-16 at 12 26 17 PM
- [ ] Click "My API Key" and copy your key
     - An API key is free to use for open-source software maintainers
- [ ] Set the environment variable on your computer
     `export APPLITOOLS_API_KEY=Your_API_Key_Here`

### Running the test

- [ ] Run the following command at the root of your project
     `npx eyes-storybook`
- [ ] In your Applitools dashboard you should see the results of your tests
Screen Shot 2019-09-16 at 12 20 46 PM

---

### Dependencies and configuration

1. Installed Eyes-Storybook as a local dev dependency:
     `yarn add --save-dev @applitools/eyes-storybook`

2. Added an `applitools.config.js` file. Below are the properties that were modified. Alternatively, this can be achieved in the command-line using `npx eyes-storybook -h localhost -p 9000 -c storybook --storybook-static-dir storybook`

|Property name                          |Value                         |Description                         |
|-------------------------------|-----------------------------|-----------------------------|
|`storybookPort`|9000            |Port to run Storybook            |
|`storybookHost`          |localhost            |Host to run Storybook            |
|`storybookConfigDir`          |storybook|Path to Storybook's config folder (default is .storybook)|
|`storybookStaticDir`          |storybook|Path to Storybook's static files folder|

> **Please note** 53 out of the 74 stories pass. It is normal for your test to take up to an hour to complete because of this. Applitools visual tests are currently running with a concurrency value of 10. This means that only up to 10 visual tests can run in parallel. Unfortunately, this is the default behavior for free accounts.

### CODE REVIEW

**Comment by tarynbillie** [comment]
@cmwhited

**Comment by cmwhited** [comment]
@nikolaglumac have you had a chance to look at this PR?

**Comment by nikolaglumac** [comment]
Sorry, @cmwhited I didn't have time so far. I will get back to you by the end of the week or at the beginning of the next one at the latest. Can you please merge the latest `develop` branch into `tarynbillie:develop` in the meantime?

**Comment by cmwhited** [comment]
Thanks for the response @nikolaglumac. @tarynbillie please see the comment and merge latest `develop` into `tarynbillie:develop`

**Comment by tarynbillie** [comment]
@nikolaglumac the latest `develop` branch is merged into `tarynbillie:develop`!

File: package.json [hunk 1]
```diff
@@ -41,6 +42,7 @@
     "electron": "./node_modules/.bin/electron"
   },
   "devDependencies": {
+    "@applitools/eyes-storybook": "^2.7.36",
```
Comment (nikolaglumac) [inline package.json]: @tarynbillie please remove `^` from the version value (we use fixed dependencies versions) and make sure yarn.lock gets updated too.

**Comment by nikolaglumac** [comment]
@DominikGuzei can you please review this one too? The code changes look good but I haven't had time to actually run Applitools...

File: package.json (hunk 1 above)
Comment (tarynbillie) [reply package.json]: These changes have been made!

**Comment by nikolaglumac** [comment]
Thanks @tarynbillie! I have updated the `yarn.lock`: https://github.com/input-output-hk/daedalus/pull/1571/commits/c972857f9279051959344b71ddc15e11ea3e64fb
The rest of the changes look good 💯 

@DominikGuzei please test Applitools and then we are ready to merge this one 🙏

**Comment by tarynbillie** [comment]
Thank you!!

**Comment by DominikGuzei** [comment]
Im reviewing this now 👍

**Comment by DominikGuzei** [comment]
@tarynbillie thanks for your work! I tested aplitools with your setup and got it running the tests (im currently re-running to test the eyes algorithm) … what is your suggested workflow for our team? I can see that the team features are all disabled for the "free" version of Aplitools. cc @cmwhited

**Comment by nikolaglumac** [comment]
@tarynbillie @cmwhited can you please answer the question @DominikGuzei posted? https://github.com/input-output-hk/daedalus/pull/1571#issuecomment-537937905
Thanks!

**Comment by nikolaglumac** [comment]
@tarynbillie @cmwhited any news about this https://github.com/input-output-hk/daedalus/pull/1571#issuecomment-537937905

**Comment by cmwhited** [comment]
@nikolaglumac sorry for the late response. I am working to write up a recommended workflow and will get that to you ASAP. Thanks for the feedback

**Comment by nikolaglumac** [comment]
Hi @cmwhited! Any news? :)
cc @DominikGuzei

**Comment by nikolaglumac** [comment]
@cmwhited we haven't heard for you for a long time. Are you facing some issues we can help you with?

**Comment by cmwhited** [comment]
Hey @nikolaglumac! Thanks for pinging me. Been crazy the last couple weeks with travel and conferences.
We like to think that this change helps to protect your repo and provides another valuable layer of tests that can be ran. For anyone who wants to run those tests, they can sign up for their own, free, API key and run those tests locally and then the OSS key can be used for CI/CD, where we see a lot of benefit.

Does that help to answer the question?

**Comment by nikolaglumac** [comment]
@DominikGuzei can you please respond to this: https://github.com/input-output-hk/daedalus/pull/1571#issuecomment-544629712
Thanks!

**Comment by DominikGuzei** [comment]
@cmwhited Thanks for the response! Where do we get the `OSS key` from? I mostly see this setup as useful in terms of automated CI

**Comment by cmwhited** [comment]
@DominikGuzei you can sign up for an OSS key through the applitools site. I agree, I think that would be the most useful in your CI. Then whoever is working on it can get a free API key from applitools to run the tests locally.

**Comment by cmwhited** [comment]
@nikolaglumac I see you added the paused label. Is there something we can do to help with this?

**Comment by nikolaglumac** [comment]
@cmwhited we are in the middle of an important release so all non-critical PRs have been paused. We will get back to this once release is out. Thanks!

**Comment by cmwhited** [comment]
@nikolaglumac thanks for the update! understood!

cc @tarynbillie

**Comment by cmwhited** [comment]
@nikolaglumac has the update been released? What does a timeline look like to remove the paused label and get some eyes on this again?

**Comment by nikolaglumac** [comment]
@cmwhited should happen next week. Thanks for patience!

**Comment by cmwhited** [comment]
@nikolaglumac thanks! Will look next week!

**Comment by cmwhited** [comment]
cc @tarynbillie

**Comment by tarynbillie** [comment]
@nikolaglumac thanks for the update!

**Comment by cmwhited** [comment]
@nikolaglumac did the release get promoted and can we get this looked at?

**Comment by nikolaglumac** [comment]
@cmwhited yes it was but we were now tied up with the next one. Sorry for the delay.
I believe we will be back to this after 10th of December 🙏

**Comment by cmwhited** [comment]
@nikolaglumac thanks for the update.

cc @tarynbillie

**Comment by cmwhited** [comment]
@nikolaglumac any update on this now that we are past the 10th?

**Comment by cmwhited** [comment]
@nikolaglumac pinging again. did the release go out? can we get some eyes on this and get it merged?

**Comment by nikolaglumac** [comment]
Hi @cmwhited!
Sorry for the delay in my response...
We are not ready to merge this just yet as there is practically a complete fork of Daedalus in `v2-integration` branch which needs to be merged into develop first. It is still not clear when this will happen but I promise to reach out to you as soon as we are ready to proceed with this PR.
Thank you for your patience!
Cheers!

**Comment by cmwhited** [comment]
@nikolaglumac Thanks for the response!

**Pull request closed without merging**